### PR TITLE
hw-mgmt: scripts: Add ASIC temperature symlinks for XDR systems

### PR DESCRIPTION
**Pull request by felixradensky**
Generate ASIC temperature and ASIC hwmon symlinks for each ASIC on XDR systems.

### CODE REVIEW

**Comment by felixradensky** [comment]
RERUN

File: usr/usr/bin/hw-management-thermal-events.sh [hunk 1]
```diff
@@ -321,8 +321,24 @@ if [ "$1" == "add" ]; then
...
 			if [ "$name" == "mlxsw" ]; then
-				ln -sf "$3$4" $cpath/asic_hwmon
-				ln -sf "$3""$4"/temp1_input "$tpath"/asic
+				case $sku in
+					HI157|HI158)
+						# Mapping of ASIC I2C bus to ASIC index
+						asic_indices=([2]=1 [18]=2 [34]=3 [50]=4)
+						asic_bus=$(echo $4 | cut -d/ -f7 | cut -d- -f2)
+						asic_index=${asic_indices[${asic_bus}]}
+						ln -sf "$3$4" $cpath/asic${asic_index}_hwmon
```
Comment (ciju-nvidia) [inline usr/usr/bin/hw-management-thermal-events.sh:330]: Can you use 'check_n_link' function for creating all links?

Comment (felixradensky) [reply usr/usr/bin/hw-management-thermal-events.sh]: Modified the patch to use check_n_link for file links

**Comment by felixradensky** [comment]
RERUN

**Review by ciju-nvidia — APPROVED** [review]

**Pull request merged**